test/e2e/basic/metrics_cert_rotation.go

```diff
@@ -0,0 +1,77 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+var _ = ginkgo.Describe("[basic][metrics] Node Tuning Operator certificate rotation", func() {
+
+	ginkgo.Context("TLS certificate rotation", func() {
+		ginkgo.It("delete certificate Secret and check that the server restarts with latest certificate", func() {
+			const (
+				pollInterval = 5 * time.Second
+				waitDuration = 5 * time.Minute
+			)
+
+			// Delete the existing secret to force certificate rotation
+			_, _, err := util.ExecAndLogCommand("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "secret/node-tuning-operator-tls")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("sleeping 5s to wait for Secret to be refreshed")
+			time.Sleep(5 * time.Second)
+
+			// Delete the secret again to force certificate rotation
```

Comment:
I didn't quite get why we're doing this twice.  Is this based on what @fcami suggested?  Can we have an explanation?

Reply:
If the fsnotify watches a file that's deleted, it will report the delete event, which might trigger reloading the certificate properly. However, rotating the certificate a second time will not work since the fsnotify does not watch the new file.
In short, testing that renewing the cert twice results in the last certificate being loaded protects against a refactoring that would simplify the fsnotify logic too much.
Does that help @jmencak ?

Reply:
It does, thank you.  I need to look into this more.  In any case, an explanation like this should go in the comments.

Reply:
I added the explanation to the second deletion.